### ISSUE TITLE
Allow extended container to use imported modules

### ISSUE THREAD

**Issue by yshrsmz** [Kodein-Framework/Kodein-DI#222]
Hi, 

`importOnce` seems to be not able to detect already imported modules in its parent Kodein. 
Is there any workaround for this?

I wrote simple test to reproduce the issue.

```kotlin
import org.kodein.di.Kodein
import org.kodein.di.erased.bind
import org.kodein.di.erased.instance
import org.kodein.di.erased.provider
import org.kodein.di.fullDescription
import kotlin.test.Test

class ditest {

    fun moduleA(): Kodein.Module {
        return Kodein.Module(name = "module-a") {
            bind<UseCaseA>() with provider { UseCaseA() }
        }
    }

    fun moduleB(): Kodein.Module {
        return Kodein.Module(name = "module-b") {
            importOnce(moduleA())
            bind<UseCaseB>() with provider { UseCaseB(useCaseA = instance()) }
        }
    }

    @Test
    fun test() {
        val container1 = Kodein {
            importOnce(moduleA())
        }

        val container2 = Kodein {
            extend(container1)
            importOnce(moduleB()) // OverridingException, as moduleA is already imported in container1
        }

        println(container2.container.tree.bindings.fullDescription())
    }
}

class UseCaseA
class UseCaseB(val useCaseA: UseCaseA)
```

this test throws following error:

```
org.kodein.di.Kodein$OverridingException: Binding bind<UseCaseA>() with ? { ? } must not override an existing binding.
	at org.kodein.di.internal.KodeinContainerBuilderImpl.checkOverrides(KodeinContainerBuilderImpl.kt:108)
	at org.kodein.di.internal.KodeinContainerBuilderImpl.bind(KodeinContainerBuilderImpl.kt:115)
	at org.kodein.di.internal.KodeinBuilderImpl$TypeBinder.with(KodeinBuilderImpl.kt:20)
	at jp.co.dotmoney.presentation.ditest$moduleA$1.invoke(ditest.kt:14)
	at jp.co.dotmoney.presentation.ditest$moduleA$1.invoke(ditest.kt:10)
	at org.kodein.di.internal.KodeinBuilderImpl.import(KodeinBuilderImpl.kt:51)
	at org.kodein.di.internal.KodeinBuilderImpl.importOnce(KodeinBuilderImpl.kt:58)
	at org.kodein.di.Kodein$Builder$DefaultImpls.importOnce$default(Kodein.kt:304)
	at jp.co.dotmoney.presentation.ditest$moduleB$1.invoke(ditest.kt:20)
	at jp.co.dotmoney.presentation.ditest$moduleB$1.invoke(ditest.kt:10)
	at org.kodein.di.internal.KodeinBuilderImpl.import(KodeinBuilderImpl.kt:51)
	at org.kodein.di.internal.KodeinBuilderImpl.importOnce(KodeinBuilderImpl.kt:58)
	at org.kodein.di.Kodein$Builder$DefaultImpls.importOnce$default(Kodein.kt:304)
	at jp.co.dotmoney.presentation.ditest$test$container2$1.invoke(ditest.kt:33)
	at jp.co.dotmoney.presentation.ditest$test$container2$1.invoke(ditest.kt:10)
	at org.kodein.di.internal.KodeinImpl$Companion.newBuilder(KodeinImpl.kt:22)
	at org.kodein.di.internal.KodeinImpl$Companion.access$newBuilder(KodeinImpl.kt:21)
	at org.kodein.di.internal.KodeinImpl.<init>(KodeinImpl.kt:19)
	at org.kodein.di.Kodein$Companion.invoke(Kodein.kt:397)
	at org.kodein.di.Kodein$Companion.invoke$default(Kodein.kt:397)
	at jp.co.dotmoney.presentation.ditest.test(ditest.kt:31)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.runTestClass(JUnitTestClassExecutor.java:110)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:58)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:38)
	at org.gradle.api.internal.tasks.testing.junit.AbstractJUnitTestClassProcessor.processTestClass(AbstractJUnitTestClassProcessor.java:62)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:51)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:32)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:93)
	at com.sun.proxy.$Proxy2.processTestClass(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.processTestClass(TestWorker.java:118)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:175)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:157)
	at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:404)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
	at java.lang.Thread.run(Thread.java:748)
```

## Environment

Kotlin 1.3.31
Kodein 6.2.1
Kotlin Multiplatform Project(Android/iOS)

**Comment by romainbsl** [comment]
Hi,

In fact, the `importOnce` function isn't made to be used in transitive container.
Its scope is limited to the current KodeinBuilder, so the following snippet would work, 
as the modules are stored into some object, thus we can know that the moduleA as already been imported. 

```kotlin
val container1 = Kodein {
    importOnce(moduleA())
    import(moduleB())
}
```

In the case of an extended container, the bindings are transitive, but not the modules. However, the output of `println(container2.container.tree.bindings.fullDescription())` is:

```
module module-a {
   bind<UseCaseA>() with provider { UseCaseA }
}
```

This is because, even if the bindings are transitively pass to the new container without there modules, we keep the origin of the binding in the binding itself. 

As a workaround, you might change your modules organization to avoid such case where `UseCaseA` isn't binded twice:

```kotlin
fun main() {
    val container1 = Kodein {
        bind<UseCaseA>() with provider { UseCaseA() }
    }

    val container2 = Kodein {
        extend(container1)
        import(moduleB())
    }

    println(container2.container.tree.bindings.fullDescription())
}

fun moduleB(): Kodein.Module {
    return Kodein.Module(name = "module-b") {
        bind<UseCaseB>() with provider { UseCaseB(useCaseA = instance()) }
    }
}

class UseCaseA
class UseCaseB(val useCaseA: UseCaseA)
```

**Comment by SalomonBrys** [comment]
Yep.
However, importOnce should work with `extends`.
We should also extend the already imported modules.
This is not the case now, but that's a valid thing to put on the roadmap :)

**Comment by romainbsl** [comment]
Will be available from `6.4`
